eth/vm/stack.py

```diff
@@ -35,6 +36,18 @@ def __init__(self) -> None:
     def __len__(self) -> int:
         return len(self.values)
 
+    def __iter__(self) -> Iterable[int]:
+        for item in self.values:
+            if isinstance(item, int):
+                yield item
+            elif isinstance(item, bytes):
+                yield big_endian_to_int(item)
+            else:
+                raise TypeError(
+                    "Invariant: stack should only contain `int` and `bytes` "
+                    "types.  Got: {0}".format(type(item))
```

Comment:
*nitpick* it can be nice to see the value in addition to the type. One way to do that would be something like:

```suggestion
                    "types.  Got: {!r}".format(item)
```

which usually is enough to determine both.